src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,251 @@
+package duke.parser;
+
+import duke.exceptions.DukeExceptionIllegalArgument;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Parser class.
+ *
+ * Handles primarily date parsing.
+ */
+public class Parser {
+
+    /**
+     * List of supported datetime formats for parsing,
+     * partially inspired by { @link https://balusc.omnifaces.org/2007/09/dateutil.html }.
+     */
+    private static final DateTimeFormatter[] DATE_FORMATS = new DateTimeFormatter[]{
+            // Time
+            // "hh a", "hh.mm a", // Same/Next day, 12-hour format, 00 minute
+            DateTimeFormatter.ofPattern("H:mm"), // Same/Next day, 24-hour format
+
+            // Date
+            DateTimeFormatter.ofPattern("d/M/yyyy"), // Full date
```

Comment:
Good effort in catering to all possible input. Perhaps what you can do is to just throw an exception if the input is not in the correct format

Reply:
They are throwing exceptions, in `parseDate` below.